pkg/api/norman/customization/cluster/validator.go

```diff
@@ -440,25 +443,73 @@ func validateEKSAccess(request *types.APIContext, eksConfig map[string]interface
 	return nil
 }
 
-func validateEKSNodegroups(request *types.APIContext, eksConfig map[string]interface{}, prevSpec *eksv1.EKSClusterConfigSpec) error {
-	noNodegroupsError := httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one nodegroup"))
-	nodegroups, ok := eksConfig["nodeGroups"].([]interface{})
-	if !ok || nodegroups == nil {
-		if request.Method == http.MethodPost {
-			return noNodegroupsError
+// validateEKSKubernetesVersion checks whether a kubernetes version is provided and if it is supported
+func validateEKSKubernetesVersion(spec *v32.ClusterSpec, prevCluster *v3.Cluster) error {
+	clusterVersion := spec.EKSConfig.KubernetesVersion
+	if clusterVersion == nil {
+		return nil
+	}
+
+	if aws.StringValue(clusterVersion) == "" {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "cluster kubernetes version cannot be empty string")
+	}
+
+	if prevCluster == nil {
+		return nil
+	}
+
+	nodegroups := prevCluster.Spec.EKSConfig.NodeGroups
+	if nodegroups == nil {
+		if prevCluster.Status.EKSStatus.UpstreamSpec == nil {
+			return httperror.NewAPIError(httperror.ClusterUnavailable, "upstream spec not available yet, unable to validate kubernetes version upgrade")
 		}
-		if prevSpec.NodeGroups != nil && len(prevSpec.NodeGroups) == 0 {
-			return noNodegroupsError
+		nodegroups = prevCluster.Status.EKSStatus.UpstreamSpec.NodeGroups
+	}
+
+	currentVersion := prevCluster.Spec.EKSConfig.KubernetesVersion
+	if currentVersion == nil {
+		currentVersion = prevCluster.Status.EKSStatus.UpstreamSpec.KubernetesVersion
+	}
```

Comment:
this idea of `a) check spec, b) check status`, is that for import ?

Reply:
Not necessarily, it just means it is not being updated. If the version is not in the request it will result in a nil value in the struct.

Reply:
Ok, that makes sense, you would still check current version if its not being updated in this request.